src/kubernetes.go

```diff
@@ -204,52 +204,57 @@ func main() {
 		}
 	}()
 
-	defer logger.Debug(exitLog)
+	defer logger.Debugf(exitLog)
 	logger.Debugf("Integration %q ver. %s (git %s) started", integrationName, integrationVersion, integrationCommitHash)
 	if args.ClusterName == "" {
-		logger.Panic(errors.New("cluster_name argument is mandatory"))
+		logger.Errorf("cluster_name argument is mandatory")
+		os.Exit(1)
 	}
 
 	nodeName := os.Getenv(nodeNameEnvVar)
 	if nodeName == "" {
-		logger.Panicf("%s env var should be provided by Kubernetes and is mandatory", nodeNameEnvVar)
+		logger.Errorf("%s env var should be provided by Kubernetes and is mandatory", nodeNameEnvVar)
+		os.Exit(1)
 	}
 
-	if !args.All && !args.Metrics {
-		return
-	}
+	// TODO: Map to v3
+	//if !args.All && !args.Metrics {
+	//	os.Exit(1)
+	//}
 
 	ttl, err := time.ParseDuration(args.DiscoveryCacheTTL)
 	if err != nil {
-		logger.WithError(err).Errorf("while parsing the cache TTL value. Defaulting to %s", defaultDiscoveryCacheTTL)
+		logger.Errorf("Error while parsing the cache TTL value, defaulting to %s: %v", defaultDiscoveryCacheTTL, err)
 		ttl = defaultDiscoveryCacheTTL
 	}
 
 	timeout := time.Millisecond * time.Duration(args.Timeout)
 
 	innerKubeletDiscoverer, err := clientKubelet.NewDiscoverer(nodeName, logger)
 	if err != nil {
-		logger.Panicf("error during Kubelet auto discovering process. %s", err)
+		logger.Errorf("Error during Kubelet auto discovering process: %v", err)
 	}
 	cacheStorage := storage.NewJSONDiskStorage(getCacheDir(discoveryCacheDir))
 
 	defaultNetworkInterface, err := network.CachedDefaultInterface(
 		logger, args.NetworkRouteFile, cacheStorage, ttl)
 	if err != nil {
-		logger.Warn(err)
+		logger.Warnf("Error finding default network interface: %v", err)
 	}
 	kubeletDiscoverer := clientKubelet.NewDiscoveryCacher(innerKubeletDiscoverer, cacheStorage, ttl, logger)
```

Comment:
Won't this panic if we use `logger.Errorf` instead of `logger.PanicF` in line 235?

Reply:
Can you mirror this comment on #163 ? Ty <3